sdk/textanalytics/Azure.AI.TextAnalytics/src/NamedEntityType.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Gets the entity type inferred by the text analytics service's named entity recognition model.
+    /// The list of available types is described at <see href="https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/named-entity-types"/>.
+    /// </summary>
+    [JsonConverter(typeof(NamedEntityTypeJsonConverter))]
+    public readonly struct NamedEntityType : IEquatable<NamedEntityType>
+    {
+        /// <summary>
+        /// Specifies that named entity contains recognized name.
+        /// </summary>
+        public static readonly NamedEntityType Person = new NamedEntityType("Person");
+
+        /// <summary>
+        /// Specifies that named entity contains natural or human-made landmarks, structures, or geographical features.JsonConvert.SerializeObject(model)
+        /// </summary>
+        public static readonly NamedEntityType Location = new NamedEntityType("Location");
+
+        /// <summary>
+        /// Specifies that named entity contains recognized name of organization, corporation, agency, or other group of people.
+        /// </summary>
+        public static readonly NamedEntityType Organization = new NamedEntityType("Organization");
+
+        /// <summary>
+        /// Specifies that named entity contains date, time or duration.
+        /// </summary>
+        public static readonly NamedEntityType DateTime = new NamedEntityType("DateTime");
+
+        /// <summary>
+        /// Specifies that named entity contains phone number (US Phone numbers only).
+        /// </summary>
+        public static readonly NamedEntityType PhoneNumber = new NamedEntityType("PhoneNumber");
+
+        /// <summary>
+        /// Specifies that named entity contains Internet URL.
+        /// </summary>
+        public static readonly NamedEntityType Email = new NamedEntityType("Email");
+
+        /// <summary>
+        /// Specifies that named entity contains Internet URL.
+        /// </summary>
+        public static readonly NamedEntityType Url = new NamedEntityType("URL");
```

Comment:
Let's change this to URI on the client side, doing whatever needs to be done to correctly serialize it from the service value.